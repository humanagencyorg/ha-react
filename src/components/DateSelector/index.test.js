import React from 'react';
import renderer from 'react-test-renderer';

import { DateSelector } from './index';
import { mount } from 'enzyme';

describe('DateSelector component', () => {
  describe('without error', () => {
    it('matches snapshot', () => {
      const props = {
        name: 'name',
        label: 'label',
      };

      const wrapper = renderer.create(<DateSelector {...props} />);

      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('with error', () => {
    it('matches snapshot', () => {
      const props = {
        name: 'name',
        label: 'label',
        error: true,
        helperText: 'error',
      };

      const wrapper = renderer.create(<DateSelector {...props} />);

      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('shows error message', () => {
      const errorMessage = 'error';
      const props = {
        error: true,
        helperText: errorMessage,
      };

      const wrapper = mount(<DateSelector {...props} />);
      const errorWrapper = wrapper.find('p.Mui-error');

      expect(errorWrapper.text()).toBe(errorMessage);
    });
  });

  it('sets default value from props', () => {
    const defaultValue = '11/11/1999';
    const props = {
      defaultValue,
    };

    const wrapper = mount(<DateSelector {...props} />);
    const input = wrapper.find('input');

    expect(input.prop('value')).toBe(defaultValue);
  });

  it('calls #onFocus from props on field focus', () => {
    const onFocus = jest.fn();
    const props = {
      onFocus,
    };

    const wrapper = mount(<DateSelector {...props} />);
    const input = wrapper.find('input');

    input.simulate('focus');

    expect(onFocus).toHaveBeenCalled();
  });
});
