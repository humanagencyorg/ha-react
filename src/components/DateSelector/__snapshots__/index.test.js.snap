// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSelector component with error matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-label-5 MuiInputLabel-formControl MuiInputLabel-shrink Mui-error Mui-error"
    data-shrink={true}
  >
    label
  </label>
  <div
    className="MuiInputBase-root MuiInput-root Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedEnd"
      disabled={false}
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
    <div
      className="MuiInputAdornment-root makeStyles-adornment-3 MuiInputAdornment-positionEnd"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <p
    className="MuiFormHelperText-root makeStyles-errorMessage-6 Mui-error"
  >
    error
  </p>
</div>
`;

exports[`DateSelector component without error matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-label-5 MuiInputLabel-formControl"
    data-shrink={false}
  >
    label
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedEnd"
      disabled={false}
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
    <div
      className="MuiInputAdornment-root makeStyles-adornment-3 MuiInputAdornment-positionEnd"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
